views/js/runner/proxy/qtiServiceProxy.js

```diff
@@ -105,7 +105,9 @@ define([
                     if (error.data && self.isConnectivityError(error.data)) {
                         self.setOffline('request');
                     }
-                    feedback().error(error);
+                    if (error instanceof Error) { //in case status 200 OK ans response.success:false we don't have Error
```

Comment:
I don't think it makes sense to check if `error instanceof Error` because ideally (from a semantic perspective), it always will be true. Since in this case we are actually breaking the semantics and trying to handle a non-Error-type rejection coming from the previous `.then()`, we could attach some unique property to the response there (L101) and check for it here.

(Also, it's questionable whether the `feedback()` is needed at all... if you are confident that all kinds of Test Runner errors are being handled in a user-friendly way (by dialogs, etc.), and they do not need to be shown as feedback errors, then you could simply remove it.)

Reply:
Removed feedback, it was added within refactoring (added core/request module, mixing old functionality of core/dataProvider/request and qtiServiceProxy) see changes in file qtiServiceProxy.js here  https://github.com/oat-sa/extension-tao-testqti/pull/1465/files
Previously there was no feedback here.

Tested on errors:
1. terminated session
2. lost connection
3. token expired